src/megaclient.cpp

```diff
@@ -9151,6 +9171,41 @@ void MegaClient::login(const byte* session, int size)
         getuserdata();
         fetchtimezone();
     }
+    else if (!session.empty() && session[0] == 2)
+    {
+        // folder link - read only or writable
+
+        CacheableReader cr(session);
+
+        byte v;
+        handle ph, rh;
+        byte k[FOLDERNODEKEYLENGTH];
+        string a1, a2, padding;
```

Comment:
Could we give meaningful names to the variables, please? :pray:

Reply:
The class already has member variables such as `publichandle` (which is what ph is short for) - keeping these variables short like that means we won't accidentally use a member variable instead of one of these temporary holding variables.   And I think the pattern is clear - for every one of these, we extract it from the string input, and if all of them are successfully extracted, then the values are assigned to the member variables that these represent.   They aren't long lived, each one is only mentioned twice (plus once to declare) and within a few lines of each other, and you can see exactly which member variable the data actually goes into in the last mention.    If we extracted those directly into the member variables, and failed halfway, the state would be a mess so it's much better to confirm extraction of all first, and then update all the member variables in an all-or-nothing "strong guarantee" fashion.
Thanks very much for the review, and for testing the changes too.
About logout, that is an interesting thought, and something I wondered about too.   However, it's possible that the folder link might be in use in multiple places (eg.  same app running on main phone and backup phone, both logged into the same link - or personal phone and work phone), and we wouldn't want logging out of one to cause the other to fail, I think.    Probably in this scenario, cancelling the link from the same place it was created (typically from the webclient) would be the way to go.

Reply:
Hi Matt. I disagree with your arguments to use these variable names. It's unclear to me what's `a1` and `a2` purpose, in example, requiring to go through `setrootnode()` and see the usage. A name like `publicHandleWriteAuth` will help readability. In case the variable name in the class clashes with the local variable's name, it's because the class member needs refactoring to `mClassMember`, with the prefix, which aims to avoid those clashes and also invalidates the point: _we won't accidentally use a member variable instead of..._
I'll apply the renaming.

As for using temporary buffers for unserializing, I agree it's the way to go. Exactly the same that is usually done when parsing data from API for many commands. No problem at all with that approach.